src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IsPatternOperator.cs

```diff
@@ -192,8 +192,6 @@ private void LowerOneTest(BoundDagTest test, bool invert = false)
                 _factory.Syntax = test.Syntax;
                 switch (test)
                 {
-                    case BoundDagAssignmentEvaluation:
```

Comment:
Consider keeping the case and throwing instead.

Reply:
We do this in LowerEvaluation which is the next thing we call.